### ISSUE TITLE
No JS injection if Browsers do-not-track option is active

### ISSUE THREAD

**Issue by resta3000** [dmind-gmbh/extension-cookieman#284]
Cookieman does not injekt Javascript after consent if do-not-track option in Browser is active.
Tested with Firefox (106) and Chrome (106).

**Comment by jonaseberle** [comment]
Yes, if it is set to do so (https://docs.typo3.org/p/dmind/cookieman/main/en-us/Configuration/Reference/TypoScriptSetup/Index.html#groups-group-key-respectdnt).
What is your configuration?
What is your expectation?

**Comment by resta3000** [comment]
I tried to change the value "respectDnt" for the associated group, but it did not do the trick.
In my scenario a user clicks an inactive map to activate it by consenting the cookie. 
In Safari there is no Problem at all, but firefox and Chrome wont fire the JS with activated doNotTrack option at all.
Only way to do it for me, was to manipulate the cookieman.js file on line 95 "return !aGroup.respectDnt || (window.navigator.doNotTrack !== '0')" – a quick and dirty way : (

**Comment by jonaseberle** [comment]
Hmm, the first part of the condition (`!aGroup.respectDnt`) does not seem to return true when the group has `respectDnt = 0` set.

https://github.com/dmind-gmbh/extension-cookieman/blob/master/Resources/Public/Js/cookieman.js#L95

For testing I have set that deliberately for the "marketing" group on https://cookieman.d-mind.de/.

**Comment by jonaseberle** [comment]
`aGroup.respectDnt` is a string ("0") at that point. But `"0"` is `true` ;)

Nice catch, thank you. It would probably be best to add the (TypoScript) string => boolean/int conversion here: https://github.com/dmind-gmbh/extension-cookieman/blob/master/Classes/DataProcessing/TypoScriptSettingsProcessor.php#L62-L73
